### PR TITLE
Offset context json

### PR DESCRIPTION
**Pull request by github-sajan**

### CODE REVIEW

File: backends/dpdk/dpdkProgramStructure.h [hunk 1]
```diff
@@ -55,6 +55,8 @@ struct DpdkProgramStructure {
     IR::Vector<IR::Type> used_metadata;
     ordered_map<cstring, std::vector<struct hdrFieldInfo>> hdrFieldInfoList;
     ordered_map<cstring, IR::ParameterList*> defActionParamList;
+    ordered_map<cstring, unsigned> keyOffsetMap;
```
Comment (kamleshbhalui) [inline backends/dpdk/dpdkProgramStructure.h]: where is it used?

**Review by kamleshbhalui — APPROVED** [review]

File: backends/dpdk/dpdkProgramStructure.h (hunk 1 above)
Comment (github-sajan) [reply backends/dpdk/dpdkProgramStructure.h]: Removed

File: backends/dpdk/dpdkAsmOpt.cpp [hunk 2]
```diff
@@ -264,5 +264,17 @@ bool ValidateTableKeys::preorder(const IR::DpdkAsmProgram *p) {
...
+// Store the user metadata structure in 'structure->mdStruct' which will be
+// later used for offset value calculation for metadata field
+bool CollectUserMetadataStructure::preorder(const IR::DpdkAsmProgram *p) {
+    for (auto st : p->structType) {
+        if (isMetadataStruct(st)) {
+            structure->mdStruct = st;
+            break;
+        }
+    }
+    return true;
```
Comment (usha1830) [inline backends/dpdk/dpdkAsmOpt.cpp]: This should be returning false. You don't have any other visitors.

**Review by usha1830 — APPROVED** [review]
Looks fine

File: backends/dpdk/dpdkAsmOpt.cpp (hunk 2 above)
Comment (github-sajan) [reply backends/dpdk/dpdkAsmOpt.cpp]: updated

**Pull request closed without merging**